### PR TITLE
graphic_functional: Add ipv6 network test

### PR DESCRIPTION
**Pull request by cliping**
Start a guest with graphic listening to ipv6 network.

Signed-off-by: Liping Cheng <lcheng@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/graphics/graphics_functional.py [hunk 1]
```diff
@@ -664,9 +666,11 @@ def get_fail_pattern(params, expected_result):
         if fail_patts or expected_spice_port == 'not_set':
             fail_patts += tls_fail_patts
 
-        for ip in expected_spice_ips:
-            if ip is not None and ip not in utils_net.get_all_ips():
-                fail_patts.append(r'binding socket to \S* failed')
+        net_type = params.get('spice_network_type')
+        if net_type != 'nat' and net_type != 'route':
```
Comment (dzhengfy) [inline libvirt/tests/src/graphics/graphics_functional.py]: if net_type not in ['nat', 'route']:

File: libvirt/tests/src/graphics/graphics_functional.py [hunk 2]
```diff
@@ -1112,6 +1120,20 @@ def setup_networks(params, test):
...
+            networks[net_type] = LibvirtNetwork(
+                net_type, address=address, net_name=net_name, persistent=True,
+                ip_version=ip_version)
+        elif net_type == 'route':
+            net_name = 'virt-test-%s' % net_type
+            ip_version = params.get('ip_version')
+            address = params.get('listen_address')
+            networks[net_type] = LibvirtNetwork(
+                net_type, address=address, net_name=net_name, persistent=True,
+                ip_version=ip_version)
```
Comment (dzhengfy) [inline libvirt/tests/src/graphics/graphics_functional.py]: Line 1131 - 1136 are same with L1124 to L1129. Let's reuse them.

**Comment by dzhengfy** [comment]
@cliping please help update accordingly. Thanks.

**Comment by cliping** [comment]
Test results:

#avocado run --vt-type libvirt graphics.functional.positive_tests.spice_only.listen_type_network
JOB ID     : 93786680058faca97df305f72a46d32f3ebdab79
JOB LOG    : /root/avocado/job-results/job-2020-04-19T23.38-9378668/job.log
 (1/5) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.spice_only.listen_type_network.network_vnet: PASS (9.20 s)
 (2/5) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.spice_only.listen_type_network.network_macvtap: CANCEL: Need to setup macvtap_device first. (2.90 s)
 (3/5) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.spice_only.listen_type_network.network_bridge: CANCEL: Need to setup bridge_device first. (2.83 s)
 (4/5) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.spice_only.listen_type_network.network_nat: PASS (18.71 s)
 (5/5) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.spice_only.listen_type_network.network_route: PASS (19.92 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 2
JOB TIME   : 55.65 s

#avocado run --vt-type libvirt graphics.functional.positive_tests.vnc_only.listen_type_network
JOB ID     : 487881c652ebecfa7c690bff7d8295ada2ed2618
JOB LOG    : /root/avocado/job-results/job-2020-04-19T23.39-487881c/job.log
 (1/5) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.vnc_only.listen_type_network.network_vnet: PASS (9.01 s)
 (2/5) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.vnc_only.listen_type_network.network_macvtap: CANCEL: Need to setup macvtap_device first. (2.89 s)
 (3/5) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.vnc_only.listen_type_network.network_bridge: CANCEL: Need to setup bridge_device first. (2.92 s)
 (4/5) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.vnc_only.listen_type_network.network_nat: PASS (19.87 s)
 (5/5) type_specific.io-github-autotest-libvirt.graphics.functional.positive_tests.vnc_only.listen_type_network.network_route: PASS (21.05 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 2
JOB TIME   : 57.96 s

**Comment by cliping** [comment]
Related PR: https://github.com/avocado-framework/avocado-vt/pull/2517

**Comment by cliping** [comment]
> Too many commits, need to merge into 1

Merge commits to one.

**Pull request merged**